mountpoint-s3/src/prefetch/checksummed_bytes.rs

```diff
@@ -123,6 +124,23 @@ pub enum IntegrityError {
     #[error("Checksum mismatch. expected: {0:?}, actual: {1:?}")]
     ChecksumMismatch(Crc32c, Crc32c),
 }
+
+/// Compare two [ChecksummedBytes].
```

Comment:
I actually did implement it! However, the meaning felt off.

There were a few options:
- Compare everything. However, it means two ChecksummedBytes that have the same content but different underlying bytes would not be considered equal.
- Compare only the byte slices relevant, but then we don't compare checksum and one could be invalid while the other is valid.

It felt strange and not like something to consider now since its only for testing, however I am happy to reconsider.

Reply:
I'd say the first option sounds right: an invalid ChecksummedBytes is not equal to a valid one with the same content.

Reply:
Sorry, I wasn't clear. In that first case, I am referring to two `ChecksummedBytes` that represent the same end slice, both with non-corrupt contents but based on different underlying bytes in memory. The `orig_bytes` and checksums would be two different pairings, but the checksum would be valid for its respective `orig_bytes`.

Reply:
Oh, sorry. I had completely misunderstood.

I see your doubts now, but all considered my preference would still be to implement `PartialEq` and compare the slices and the "integrity states".

Reply:
Discussed offline, let's implement `PartialEq` but only for tests.

There's weird behaviour where if we compare the validity, what should we do with it? For testing-only implementation, let's go ahead and panic.